src/main/java/com/sparta/ezpzuser/domain/popup/repository/popup/PopupRepositoryCustomImpl.java

```diff
@@ -0,0 +1,62 @@
+package com.sparta.ezpzuser.domain.popup.repository.popup;
+
+import com.querydsl.core.types.Expression;
+import com.querydsl.core.types.dsl.BooleanExpression;
+import com.querydsl.core.types.dsl.Wildcard;
+import com.querydsl.jpa.impl.JPAQuery;
+import com.querydsl.jpa.impl.JPAQueryFactory;
+import com.sparta.ezpzuser.domain.popup.dto.PopupCondition;
+import com.sparta.ezpzuser.domain.popup.entity.Popup;
+import com.sparta.ezpzuser.domain.popup.enums.ApprovalStatus;
+import com.sparta.ezpzuser.domain.popup.enums.PopupStatus;
+import lombok.RequiredArgsConstructor;
+import org.springframework.data.domain.Page;
+import org.springframework.data.domain.Pageable;
+import org.springframework.data.support.PageableExecutionUtils;
+
+import java.util.List;
+import java.util.Objects;
+
+import static com.sparta.ezpzuser.domain.popup.entity.QPopup.popup;
+
+@RequiredArgsConstructor
+public class PopupRepositoryCustomImpl implements PopupRepositoryCustom {
+
+    private final JPAQueryFactory jpaQueryFactory;
+
+    @Override
+    public Page<Popup> findAllPopupsByStatus(Pageable pageable, PopupCondition cond) {
+        JPAQuery<Popup> query = findAllPopupsByStatusQuery(popup, cond)
+                .offset(pageable.getOffset())
+                .limit(pageable.getPageSize())
+                .orderBy(popup.createdAt.desc());
+
+        List<Popup> popups = query.fetch();
+        Long totalSize = countQuery(cond).fetch().get(0);
+
+        return PageableExecutionUtils.getPage(popups, pageable, () -> totalSize);
+    }
+
+    private <T> JPAQuery<T> findAllPopupsByStatusQuery(Expression<T> expr, PopupCondition cond) {
+        return jpaQueryFactory.select(expr)
+                .from(popup)
+                .where(
+                        popup.approvalStatus.eq(ApprovalStatus.APPROVED),
+                        popupStatusEq(cond.getPopupStatus())
+                );
+    }
+
+    private JPAQuery<Long> countQuery(PopupCondition cond) {
+        return jpaQueryFactory.select(Wildcard.count)
+                .from(popup)
+                .where(
+                        popup.approvalStatus.eq(ApprovalStatus.APPROVED),
+                        popupStatusEq(cond.getPopupStatus())
+                );
+    }
```

Comment:
메서드가 너무 분리되어 있어서 가독성이 떨어지는 것 같아요. 다음과 같이 변경해보는건 어때요??
```suggestion
    private final JPAQueryFactory queryFactory;

    @Override
    public Page<Popup> findAllPopupsByStatus(Pageable pageable, String popupStatus) {
        // 데이터 조회 쿼리
        List<Popup> popups = queryFactory
                .selectFrom(popup)
                .where(
                        popup.approvalStatus.eq(ApprovalStatus.APPROVED),
                        popupStatusEq(popupStatus)
                )
                .offset(pageable.getOffset())
                .limit(pageable.getPageSize())
                .orderBy(popup.createdAt.desc())
                .fetch();

        // 전체 카운트 쿼리
        Long totalSize = queryFactory
                .select(Wildcard.count)
                .from(popup)
                .where(
                        popup.approvalStatus.eq(ApprovalStatus.APPROVED),
                        popupStatusEq(popupStatus)
                )
                .fetchOne();

        return PageableExecutionUtils.getPage(popups, pageable, () -> RepositoryUtil.getTotal(totalCount));
    }
```

Reply:
수정하겠습니다.